parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -330,36 +354,55 @@ public static Binary fromByteArray(final byte[] value) {
   }
 
   private static class ByteBufferBackedBinary extends Binary {
```

Comment:
I was thinking this, but I know that binary is used to wrap each value read out of a variable length column. I was thinking it might be worth trying to avoid the extra object allocation. It would make the code more concise though.